sys/include/crypto/sha256.h

```diff
@@ -100,6 +100,23 @@ void sha256_final(unsigned char digest[32], sha256_context_t *ctx);
  */
 unsigned char *sha256(const unsigned char *d, size_t n, unsigned char *md);
 
+/**
+ * @brief function to compute a hmac-sha256 from a given message
+ *
+ * @param[in] key key used in the hmac-sha256 computation
+ * @param[in] key_length the size in bytes of the key
+ * @param[in] message poiter to the message to generate the hmac-sha256
+ * @param[in] message_length the length of the message in bytes
+ * @param[out] result the computed hmac-sha256,
+ *             length MUST be SHA256_DIGEST_LENGTH
+ *             if result == NULL, a static buffer is used
+ */
+unsigned char * hmac_sha256(const unsigned char *key,
+                            size_t key_length,
+                            const unsigned *message,
```

Comment:
Its somehow intended.
I want to keep it consistent with the undocumented behaviour of the `sha256` function.
But now that you pointed me I'm not really happy with this thread-unsafe behaviour, neither in `sha256` nor here.